rudder-agent/SOURCES/rudder-agent-postinst

```diff
@@ -91,85 +84,18 @@ fi
 # Copy CFEngine binaries (Changed location from sbin/ to bin/ in version 3.4)
 cp -r -f /opt/rudder/bin/cf-* ${CFE_DIR}/bin/
 cp -r -f /opt/rudder/bin/rpmvercmp ${CFE_DIR}/bin/
-NB_COPIED_BINARIES=`ls -1 ${CFE_DIR}/bin/ | wc -l`
-if [ ${NB_COPIED_BINARIES} -gt 0 ]; then
-  echo "CFEngine binaries copied to workdir" >> ${LOG_FILE}
-fi
-
-# Backup rudder-server-roles.conf
-if [ -f ${CFE_DIR}/inputs/rudder-server-roles.conf ]
-then
-  mkdir -p /var/backups/rudder
-  cp -r ${CFE_DIR}/inputs/rudder-server-roles.conf /var/backups/rudder/
-  RESTORE_SERVER_ROLES_BACKUP=1
-fi
-
-# Copy initial promises if there aren't any policies already,or
-# if the cf-promises validation fails.
-RUDDER_UUID=`cat /opt/rudder/etc/uuid.hive 2>>${LOG_FILE} || true`
-if [ ! -f ${CFE_DIR}/inputs/promises.cf ] || ! ${CFE_DIR}/bin/cf-promises >> ${LOG_FILE} 2>&1
-then
-  mkdir -p ${CFE_DIR}/inputs
-  rm -rf ${CFE_DIR}/inputs/* || true
-  mkdir -p /var/rudder/ncf/common
-  rm -rf /var/rudder/ncf/common/*
-fi
-
-# Restore rudder-server-roles.conf if necessary
-if [ "${RESTORE_SERVER_ROLES_BACKUP}" = "1" ]; then
-  cp -r /var/backups/rudder/rudder-server-roles.conf ${CFE_DIR}/inputs/rudder-server-roles.conf
-fi
-
-printf "INFO: Ensuring correct permissions on the keys directory..."
-# The chmod is a workaround for the limitations of rpm in rhel6 but it could be useful in other situations
-chmod 700 ${CFE_DIR}/ppkeys
-echo " Done."
-
-# Create a key if we don't have one yet
-if [ ! -f ${CFE_DIR}/ppkeys/localhost.priv ]
-then
-  printf "INFO: Creating keys for Rudder agent..."
-  ${CFE_DIR}/bin/cf-key -T 4096 >> ${LOG_FILE} 2>&1
-  echo " Done."
-fi
-
-# Remove cfengine lock log file : http://www.rudder-project.org/redmine/issues/5488
-rm -f ${CFE_DIR}/cf3.*.runlog*
-
-# Remove the lock on CFEngine
-if [ ${I_SET_THE_LOCK} -eq 1 ]; then
-  ${RUDDER_CMD} agent enable
-fi
 
 # Ensure the modification date of the capability file is correct
 # as it is used for cache invalidation in ncf lis-compatible-inputs
 touch /opt/rudder/etc/agent-capabilities
 
-# Restart daemons if we stopped them, otherwise not
```

Comment:
Shouldn't we restart the service in postinst anyway?

Reply:
i don't know, why should we ?

Reply:
and the code below didn't restart but just started it

Reply:
to use the new cf-serverd and cf-execd binaries at least I think.

Reply:
what if we stop it and let agent-check restart it ?

Reply:
possible too